File: python/283_moveZeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        nonzero = 0
        for i in nums:
            if i != 0:
                nums[nonzero] = i
                nonzero += 1
                
        nums[nonzero:] = [0] * (len(nums) - nonzero)

File: C/283_moveZeroes.c
void moveZeroes(int* nums, int numsSize) {
    int nonzero = 0;
    int i;
    for (i = 0; i < numsSize; i++) {
        if (nums[i] != 0) {
            nums[nonzero] = nums[i];
            nonzero++;
        }
    }

    for (;nonzero < numsSize; nonzero++) {
        nums[nonzero] = 0;
    }
}
